From f3063294fa470bf5bf6e48cfbcd4dd77fa46f5ec Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Tue, 11 Apr 2023 00:32:47 +0300
Subject: [PATCH 2/7] [compiler-rt] [test] [builtins] Set
 __COMPAT_LAYER=RunAsInvoker when running tests

Set __COMPAT_LAYER=RunAsInvoker to avoid the error
"[WinError 740] The requested operation requires elevation";
heuristics can flag "trampoline_setup_test" as an executable that
may need to be run with elevated privileges.

This is the same fix as D137772 and D148988, but applied on another
set of tests.

Differential Revision: https://reviews.llvm.org/D149004
---
 compiler-rt/test/builtins/Unit/lit.cfg.py | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/compiler-rt/test/builtins/Unit/lit.cfg.py b/compiler-rt/test/builtins/Unit/lit.cfg.py
index e3602a99087d..81f01cec2a3b 100644
--- a/compiler-rt/test/builtins/Unit/lit.cfg.py
+++ b/compiler-rt/test/builtins/Unit/lit.cfg.py
@@ -128,3 +128,12 @@ if len(builtins_source_feature_duplicates) > 0:
       builtins_source_feature_duplicates)
   )
 config.available_features.update(builtins_source_features)
+
+# Avoid Windows heuristics which try to detect potential installer programs
+# (which may need to run with elevated privileges) and ask if the user wants
+# to run them in that way. This heuristic may match for executables containing
+# the word "update", e.g. `TestCases/asan_update_allocation.cpp`. Set an
+# environment variable indicating that we want to execute them with the current
+# user.
+if config.host_os == 'Windows':
+  config.environment['__COMPAT_LAYER'] = 'RunAsInvoker'
-- 
2.34.1

